### PR TITLE
WIP: Parquet and FileSystem datastore

### PR DESCRIPTION
**Pull request by jahhulbert-ccri**

### CODE REVIEW

File: build/cqs.tsv [hunk 1]
```diff
@@ -220,6 +226,7 @@ org.scalatra:scalatra-swagger_2.11	2.3.0	compile
 org.slf4j:slf4j-api	1.7.5	compile
 org.slf4j:slf4j-log4j12	1.7.5	compile
 org.tukaani:xz	1.0	compile
+org.xerial.snappy:snappy-java	1.1.1.6	compile
```
Comment (elahrvivaz) [inline build/cqs.tsv]: we'll need to exclude this...

File: geomesa-fs/geomesa-fs-datastore/src/test/scala/org/locationtech/geomesa/fs/FileSystemDataStoreTest.scala [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+@RunWith(classOf[JUnitRunner])
+class FileSystemDataStoreTest extends Specification {
+
+  sequential
+
+  "FileSystemDataStore" should {
+    "pass a test" >> {
+      // TODO use random dir path
+      val dir = new java.io.File("/tmp/awetmpawet")
```
Comment (jnh5y) [inline geomesa-fs/geomesa-fs-datastore/src/test/scala/org/locationtech/geomesa/fs/FileSystemDataStoreTest.scala]: Yeah, this is the combination to my luggage.

File: build/cqs.tsv (hunk 1 above)
Comment (jahhulbert-ccri) [inline build/cqs.tsv]: k

File: geomesa-fs/geomesa-fs-datastore/src/test/scala/org/locationtech/geomesa/fs/FileSystemDataStoreTest.scala (hunk 2 above)
Comment (jahhulbert-ccri) [inline geomesa-fs/geomesa-fs-datastore/src/test/scala/org/locationtech/geomesa/fs/FileSystemDataStoreTest.scala]: hehehe there's a todo!

File: geomesa-tools/src/main/scala/org/locationtech/geomesa/tools/ingest/AbstractIngest.scala [hunk 3]
```diff
@@ -70,6 +70,24 @@ abstract class AbstractIngest(val dsParams: Map[String, String],
 
   val ds = DataStoreFinder.getDataStore(dsParams)
 
+  // TODO modify to use dual callbacks...now we have reduce phases
+  case class MapOnlyCallback(startTime: Long,
```
Comment (elahrvivaz) [inline geomesa-tools/src/main/scala/org/locationtech/geomesa/tools/ingest/AbstractIngest.scala]: should we remove these for now, or are you planning on implementing something for this pr?

File: geomesa-fs/geomesa-fs-storage-common/src/main/scala/org/locationtech/geomesa/fs/storage/common/FileMetadata.scala [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    parts.foreach { p => out.writeBytes(p); out.write('\n') }
+    out.hflush()
+    out.hsync()
+    out.close()
+    cached = null
+    logger.info(s"wrote ${parts.size} partitions to metadata file")
+  }
+
+  override def getPartitions: java.util.List[String] = {
+    if (cached == null) cached = load()
```
Comment (jahhulbert-ccri) [inline geomesa-fs/geomesa-fs-storage-common/src/main/scala/org/locationtech/geomesa/fs/storage/common/FileMetadata.scala]: probs

File: geomesa-tools/src/main/scala/org/locationtech/geomesa/tools/ingest/AbstractIngest.scala (hunk 3 above)
Comment (jahhulbert-ccri) [inline geomesa-tools/src/main/scala/org/locationtech/geomesa/tools/ingest/AbstractIngest.scala]: yeah gonna pull it out

**Comment by jahhulbert-ccri** [comment]
closing for another fcr PR

**Pull request closed without merging**